dicetable.py

```diff
@@ -0,0 +1,327 @@
+    if you int(the string) it's legal input for python'''
+    string = str(num)
+    power = str(len(string)-1)
+    digits = string[0]+'.'+string[1:4]
+    return digits+'e+'+power
+
+
+def makealist():
+    '''take user input of a range of numbers
+    and outputs (list of numbers,input string)
+    a helper function for getstat'''
+    x = raw_input('enter number(s) you want stats of separated by commas or you can use a dash \nfor a range of numbers: ')
+    if x == 'q' or x == '':
+        return None,None
+    out = []
+    try:
+        for el in x.split(','):
+            if '-' in el:
+                start = int(el.split('-')[0])
+                end = int(el.split('-')[1])
+                if start <= end:
+                    for num in range(start,end+1):
+                        out.append(num)
+                else:
+                    for num in range(end,start+1):
+                        out.append(num)   
+            else:
+                out.append(int(el))
+        
+    except ValueError:
+        print
+        print 'ooops.  looks like someone can\'t follow directions.'
+        print 'your prize is a list that includes 0 and some of your elements.'
+        print 'here ya go.'
+        print
+        out.append(0)
+    return out,x
+
+
+#i need to modify the overflowerror, maybe use a function like scinote. 
+#right now, since overflowerror just uses int the chance will say 1 in 2 when pct is 34-50
+def getstat(dtable):
+        '''returns the stats on lst of elements in self'''
+        totalval = 0
+        tabletotal = dtable.dsize**dtable.totaldice
+        #below takes the values from makealist to get lst for function and 
+        # str to print the range of values
+        lst,lststring = makealist()
+        if lst == None:
+            return None
+        
+        for el in lst:
+            totalval += dtable.table.get(el,0)
+        if totalval == 0:
+            print 'you get nothing!  good day, sir!'
+            return 0
+        
+        
+        else:
+            
+            try:
+                chance = round(float(tabletotal)/totalval,3)
+            except OverflowError:
+                #!!personal reminder!!
+                #this needs revision.  33-50pct are all 1 in 2 chance for very large numbers
+                #all because of this line of code perhaps use funtion like scinote
+                #pass a short float and power of ten.  can use in next line for better pct, too
+                #end !!personal reminder!!
+                chance = (tabletotal)/totalval
+            
+            try:
+                pct = round(float(totalval)*100/tabletotal,3)
+            except OverflowError:
+                pct = (totalval)*100/tabletotal
+            
+            #the if elses establish output str based on size of numbers
+            #either just display the num, or display it in scientific notation if too big
+            if totalval>1000000:
```

Comment:
This looks like a magic number.
https://en.wikipedia.org/wiki/Magic_number_(programming)